multimodal/src/autogluon/multimodal/utils/object_detection.py

```diff
@@ -658,12 +725,379 @@ def get_coco_format_classes(sample_data_path):
+        tuple: top left corner of the text region, tl = (x1, y1)
+    bg_color
+        tuple: the color of the background, e.g. (0, 0, 0)
+    alpha
+        float: the opacity of the background
+    font
+        the font of the text, e.g. cv2.FONT_HERSHEY_DUPLEX
+    text_scale
+        float: the scale (font size) of the text, e.g. 0.75
+    text_thickness
+        int: the font weight of the text, e.g. 1
+    text_vert_padding
+        int: vertical padding of the text on each side
+    Returns
+    -------
+    an np.ndarray of image with added text
+    """
+    try:
+        import cv2
+    except:
+        raise ImportError("No module named: cv2. Please install cv2 by 'pip install cv2'")
+
+    x1, y1 = tl
+
+    overlay = im.copy()
+    text_size, _ = cv2.getTextSize(text, font, float(text_scale), text_thickness)
+    text_w, text_h = text_size
+
+    text_vert_padding = text_vert_padding if text_vert_padding else int(text_h * 0.1)
+
+    y1 = max(y1 - text_h - text_vert_padding * 2, 0)
+
+    cv2.rectangle(overlay, (x1, y1), (x1 + text_w, y1 + text_h + text_vert_padding * 2), bg_color, -1)
+    im = cv2.addWeighted(overlay, alpha, im, 1 - alpha, 0)
+    cv2.putText(
+        im, text, (x1, y1 + text_h + text_vert_padding), font, text_scale, (255, 255, 255), thickness=text_thickness
+    )
+    return im
+
+
+def get_color(idx):
+    idx = idx * 3
+    color = ((37 * idx) % 255, (17 * idx) % 255, (29 * idx) % 255)
+    return color
+
+
+def save_result_df(pred: Iterable, data: Union[pd.DataFrame, Dict], result_path: str, detection_classes: List[str]):
+    """
+    Saving detection results in pd.DataFrame format (per image)
+
+    Parameters
+    ----------
+    pred
+        List containing detection results for one image
+    data
+        pandas data frame or dict containing the image information to be tested
+    result_path
+        path to save result
+    detection_classes
+        all available classes for this detection
+    Returns
+    -------
+    None
+    """
+    if isinstance(data, dict):
+        image_names = data["image"]
+    else:
+        image_names = data["image"].to_list()
+    results = []
+    idx2classname = {i: classname for (i, classname) in enumerate(detection_classes)}
+
+    for image_pred, image_name in zip(pred, image_names):
+        box_info = []
+        for class_idx, bboxes in enumerate(image_pred):
+            pred_class = idx2classname[class_idx]
+
+            for bbox in bboxes:
+                box_info.append({"class": pred_class, "bbox": list(bbox[:4]), "score": bbox[4]})
+        results.append([image_name, box_info])
+    result_df = pd.DataFrame(results, columns=["image", "bboxes"])
+    result_df.to_csv(result_path, index=False)
+    logger.info("Saved detection results to {}".format(result_path))
+    print("Saved detection results to {}".format(result_path))
+
+
+def save_result_df_deprecated(pred: Iterable, data: pd.DataFrame, result_path: str, detection_data_path: str = None):
```

Comment:
If `save_result_df` can work properly, we might want to delete this function directly. Is there any code actually using `save_result_df_deprecated`?

Reply:
oh I forgot to delete this. I will delete it now.